tests/unit-tests/api/data.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Tests for Data API.
+ *
+ * @package WooCommerce\Tests\API
+ * @since 3.1.0
+ */
+
+class Data_API extends WC_REST_Unit_Test_Case {
+	/**
+	 * Setup our test server and user info.
+	 */
+	public function setUp() {
+		parent::setUp();
+		$this->user = $this->factory->user->create( array(
+			'role' => 'administrator',
+		) );
+	}
+
+	/**
+	 * Test route registration.
+	 *
+	 * @since 3.1.0
+	 */
+	public function test_register_routes() {
+		$routes = $this->server->get_routes();
+		$this->assertArrayHasKey( '/wc/v2/data', $routes );
+		$this->assertArrayHasKey( '/wc/v2/data/locations', $routes );
+	}
+
+	/**
+	 * Test getting the data index.
+	 * @since 3.1.0
+	 */
+	public function test_get_index() {
+		wp_set_current_user( $this->user );
+		$response = $this->server->dispatch( new WP_REST_Request( 'GET', '/wc/v2/data' ) );
+		$index = $response->get_data();
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( 2, count( $index ) );
+		$this->assertEquals( 'locations', $index[0]['slug'] );
+	}
+
+	/**
+	 * Test getting locations.
+	 * @since 3.1.0
+	 */
+	public function test_get_locations() {
+		wp_set_current_user( $this->user );
+		$response = $this->server->dispatch( new WP_REST_Request( 'GET', '/wc/v2/data/locations' ) );
+		$locations = $response->get_data();
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertTrue( is_array( $locations ) );
+		$this->assertGreaterThan( 1, count( $locations ) );
+		$this->assertTrue( isset( $locations[0]['code'] ) );
```

Comment:
We can use `assertNotEmpty( $locations[0]['code'])` here instead, and on other similar checks.